core/trino-main/src/main/java/io/trino/FeaturesConfig.java

```diff
@@ -92,7 +93,7 @@ public class FeaturesConfig
     /**
      * default value is overwritten for fault tolerant execution in {@link #applyFaultTolerantExecutionDefaults()}}
      */
-    private CompressionCodec exchangeCompressionCodec = NONE;
+    private CompressionCodec exchangeCompressionCodec = ZSTD;
```

Comment:
This could have a significant performance impact. I would decouple this from the change to enable HTTP/2 by default./

Reply:
@martint I've pushed just to benchmark this along with other changes :) I'll drop it before merging.

Reply:
I'm hoping that native (de)compression will be fast enough that it won't incur additional latency.

Reply:
i don't think it can be so?

Reply:
Actually it doesn't work at all ;-) It seems that it's broken - some of the tests are failing.

Reply:
> I'm hoping that native (de)compression will be fast enough that it won't incur additional latency.

It's not just about latency, but CPU required to compress and decompress. ZSTD is fast, but it's not free.